src/system.ts

```diff
@@ -91,8 +92,13 @@ export async function prepareSystem({
   server.forGet(pacFilePath).thenReply(200, pacFileContents, {
     "content-type": "application/x-ns-proxy-autoconfig",
   });
-  // TODO: Cross Platform Support, and error handling.
-  execSync(
-    `networksetup -setautoproxyurl "Wi-Fi" "https://localhost:${proxyPort}${pacFilePath}"`
-  );
+
+  get_active_interface((_, { desc: InterfaceName }) => {
```

Comment:
This does fail for me intially, edge case we can fix later.

```
[👴🏻 Malcolm] - Ready and Willing to Serve! (https://localhost:6969)
Configuring [Thunderbolt Ethernet Slot 0]...
node:child_process:965
    throw err;
    ^

Error: Command failed: networksetup -setautoproxyurl "Thunderbolt Ethernet Slot 0" "https://localhost:6969/proxy.pac"
    at genericNodeError (node:internal/errors:984:15)
    at wrappedFn (node:internal/errors:538:14)
    at checkExecSyncError (node:child_process:890:11)
    at execSync (node:child_process:962:15)
    at <anonymous> (/Users/hacksore/code/opensource/malcolm/src/system.ts:100:5)
    at /Users/hacksore/code/opensource/malcolm/node_modules/network/lib/index.js:119:11
    at /Users/hacksore/code/opensource/malcolm/node_modules/network/lib/darwin.js:13:7
    at ChildProcess.exithandler (node:child_process:414:7)
    at ChildProcess.emit (node:events:518:28)
    at ChildProcess.emit (node:domain:488:12) {
  status: 4,
  signal: null,
  output: [
    null,
```

Reply:
I am curious why this happens, can we troubleshoot this? Perhaps a bug in `network` package, or my use of `get_active_interface`